hamilton/plugins/polars_extensions.py

```diff
@@ -29,3 +37,186 @@ def register_types():
+            kwargs["missing_utf8_is_empty_string"] = self.missing_utf8_is_empty_string
+        if self.ignore_errors is not None:
+            kwargs["ignore_errors"] = self.ignore_errors
+        if self.try_parse_dates is not None:
+            kwargs["try_parse_dates"] = self.try_parse_dates
+        if self.n_threads is not None:
+            kwargs["n_threads"] = self.n_threads
+        if self.infer_schema_length is not None:
+            kwargs["infer_schema_length"] = self.infer_schema_length
+        if self.batch_size is not None:
+            kwargs["batch_size"] = self.batch_size
+        if self.n_rows is not None:
+            kwargs["n_rows"] = self.n_rows
+        if self.encoding is not None:
+            kwargs["encoding"] = self.encoding
+        if self.low_memory is not None:
+            kwargs["low_memory"] = self.low_memory
+        if self.rechunk is not None:
+            kwargs["rechunk"] = self.rechunk
+        if self.use_pyarrow is not None:
+            kwargs["use_pyarrow"] = self.use_pyarrow
+        if self.storage_options is not None:
+            kwargs["storage_options"] = self.storage_options
+        if self.skip_rows_after_header is not None:
+            kwargs["skip_rows_after_header"] = self.skip_rows_after_header
+        if self.row_count_name is not None:
+            kwargs["row_count_name"] = self.row_count_name
+        if self.row_count_offset is not None:
+            kwargs["row_count_offset"] = self.row_count_offset
+        if self.sample_size is not None:
+            kwargs["sample_size"] = self.sample_size
+        if self.eol_char is not None:
+            kwargs["eol_char"] = self.eol_char
+        if self.raise_if_empty is not None:
+            kwargs["raise_if_empty"] = self.raise_if_empty
+        return kwargs
+
+    def load_data(self, type_: Type) -> Tuple[DATAFRAME_TYPE, Dict[str, Any]]:
+        df = pl.read_csv(self.file, **self._get_loading_kwargs())
+        metadata = utils.get_file_metadata(self.file)
+        return df, metadata
+
+    @classmethod
+    def name(cls) -> str:
+        return "csv"
+
+
+@dataclasses.dataclass
+class PolarsCSVWriter(DataSaver):
+    """Class specifically to handle saving CSV files with Polars.
+    Should map to https://pola-rs.github.io/polars/py-polars/html/reference/api/polars.DataFrame.write_csv.html
+    """
+
+    file: Union[BytesIO, TextIOWrapper, str, Path]
+    # kwargs:
+    has_header: bool = True
+    separator: str = ","
+    line_terminator: str = "\n"
+    quote: str = '"'
+    batch_size: int = 1024
+    datetime_format: str = None
+    date_format: str = None
+    time_format: str = None
+    float_precision: int = None
+    null_value: str = None
+    quote_style: CsvQuoteStyle = None
+
+    @classmethod
+    def applicable_types(cls) -> Collection[Type]:
+        return [DATAFRAME_TYPE]
+
+    def _get_saving_kwargs(self):
+        kwargs = {}
+        if self.separator is not None:
```

Comment:
You could do this with a for-loop but I guess copilot does this cleanly without one :)

Reply:
Would be nice to have a function wrapper annotation, but it requires types on their part